tests/utils.py

```diff
@@ -172,21 +172,38 @@ def isnan(x: Any):
                 return False
 
         def eq(left, right) -> bool:
-            if isinstance(left, pd.DataFrame):
-                if left.columns != right.columns:
-                    return False
-            if isinstance(left, pd.Series):
-                if left.name != right.name:
-                    return False
+            if type(left) != type(right):
```

Comment:
This `if` condition is new. Question without context: is this inferred from the old checks?

Reply:
short answer: yes.

long answer: `the truth value of an array is ambiguous` is quite hard to work around without stuff like this.

To make matters worse, potential `nan` values may be in the array, but may also be plain values. basically, if it is an array we need to check that all its components are either identical or nan both left and right. if it's anything else, if it is nan, then check that they both are. if not, they need to be equal. Just explaining this logic makes me want to puke but it is the way it is.